component/distributed_cache/distributed_cache.go

```diff
@@ -365,39 +366,75 @@ func (dc *DistributedCache) GetAttr(options internal.GetAttrOptions) (*internal.
 }
 
 func (dc *DistributedCache) StreamDir(options internal.StreamDirOptions) ([]*internal.ObjAttr, string, error) {
+	var dirList []*internal.ObjAttr
+	var token string
+	var err error
+
 	isAzurePath, isDcachePath, isDebugPath, rawPath := getFS(options.Name)
 
 	if isDcachePath {
 		log.Debug("DistributedCache::StreamDir : Path is having Dcache subcomponent, path : %s", options.Name)
 		rawPath = filepath.Join(mm.GetMdRoot(), "Objects", rawPath)
+		options.Name = rawPath
+		if dirList, token, err = dc.NextComponent().StreamDir(options); err != nil {
+			return dirList, token, err
+		}
+		dirList = parseDcacheMetadataForDirEntries(dirList)
 	} else if isAzurePath {
 		log.Debug("DistributedCache::StreamDir : Path is having Azure subcomponent, path : %s", options.Name)
+		options.Name = rawPath
+		if dirList, token, err = dc.NextComponent().StreamDir(options); err != nil {
+			return dirList, token, err
+		}
 	} else if isDebugPath {
 		log.Debug("DistributedCache::StreamDir : Path is having Debug subcomponent, path : %s", options.Name)
 		return debug.StreamDir(options)
 	} else {
 		//
 		// Semantics for Readdir for unquailified path, If the dirent exist in both Dcache and Azure filesystem, then dirent
-		// present in the dcache takes the precedence.
+		// present in the dcache takes the precedence over azure, also both the entries would be listed.
+		// To know which virtual fs we are currently in we use options.IsFsDcache.
 		//
-		// TODO: Implement this, rn serving from Azure.
-		common.Assert(rawPath == options.Name, rawPath, options.Name)
-	}
+		if *options.IsFsDcache { // List from dcache.
+			log.Debug("DistributedCache::StreamDir : Listing on Unquailified path, listing from dcache, path : %s", options.Name)
+			rawPath = filepath.Join(mm.GetMdRoot(), "Objects", rawPath)
+			options.Name = rawPath
+			if dirList, token, err = dc.NextComponent().StreamDir(options); err != nil {
+				return dirList, token, err
+			}
+			dirList = parseDcacheMetadataForDirEntries(dirList)
+			for _, attr := range dirList {
+				options.DcacheEntries[attr.Name] = struct{}{}
+			}
+			if token == "" {
+				// Now start listing from the Azure.
+				*options.IsFsDcache = false
+				token = dcacheDirContToken
+			}
```

Comment:
pls leave a new line after every block., so

```
			rawPath = filepath.Join(mm.GetMdRoot(), "Objects", rawPath)
			options.Name = rawPath
			if dirList, token, err = dc.NextComponent().StreamDir(options); err != nil {
				return dirList, token, err
			}

			dirList = parseDcacheMetadataForDirEntries(dirList)
			for _, attr := range dirList {
				options.DcacheEntries[attr.Name] = struct{}{}
			}

			if token == "" {
				// Empty token signifies end-of-directory for dcache listing, start listing from Azure.
				// We set token to the special non-empty value to prevent fuse from treating this as
				// end-of-directory.
				*options.IsFsDcache = false
				token = dcacheDirContToken
			}
```

pls copy the above as it has a comment suggestion too